File: lib/core/models/master_product.dart
class MasterProduct {
  final String id;
  final String name;
  double totalStockKg;

  MasterProduct({
    required this.id,
    required this.name,
    this.totalStockKg = 0.0,
  });

  Map<String, dynamic> toMap() {
    return {
      'id': id,
      'name': name,
      'totalStockKg': totalStockKg,
    };
  }

  factory MasterProduct.fromMap(Map<String, dynamic> map) {
    return MasterProduct(
      id: map['id'] ?? '',
      name: map['name'] ?? '',
      totalStockKg: (map['totalStockKg'] as num?)?.toDouble() ?? 0.0,
    );
  }
}

File: backend/models/MasterProduct.js
const mongoose = require('mongoose');

const masterProductSchema = new mongoose.Schema({
  _id: { type: String, required: true },
  name: { type: String, required: true },
  totalStockKg: { type: Number, default: 0.0 },
}, { timestamps: true });

masterProductSchema.set('toJSON', {
  virtuals: true,
  transform: (doc, ret) => {
    ret.id = ret._id;
    delete ret._id;
    delete ret.__v;
    return ret;
  }
});

const MasterProduct = mongoose.model('MasterProduct', masterProductSchema);
module.exports = MasterProduct;
